diff --git a/src/build/SharedDeps.zig b/src/build/SharedDeps.zig
index 7f777faa1..ff218c9e5 100644
--- a/src/build/SharedDeps.zig
+++ b/src/build/SharedDeps.zig
@@ -901,6 +901,10 @@ pub fn addSimd(
             .optimize = optimize,
             .no_libcxx = true,
         })) |simdutf_dep| {
+            // Propagate the vendored simdutf headers to consumers compiling
+            // Ghostty's SIMD C++ sources under their own module.
+            m.addIncludePath(simdutf_dep.path("vendor"));
+
             m.linkLibrary(simdutf_dep.artifact("simdutf"));
             if (static_libs) |v| try v.append(
                 b.allocator,
